src/main/java/org/opensearch/ad/NodeStateManager.java

```diff
@@ -363,21 +362,22 @@ public void setException(String detectorId, Exception e) {
             return;
         }
         NodeState state = states.computeIfAbsent(detectorId, d -> new NodeState(detectorId, clock));
-        Optional<AnomalyDetectionException> exception = state.getException();
+        Optional<Exception> exception = state.getException();
         if (exception.isPresent()) {
             Exception higherPriorityException = ExceptionUtil.selectHigherPriorityException(e, exception.get());
             if (higherPriorityException != e) {
                 return;
             }
         }
 
-        AnomalyDetectionException adExep = null;
-        if (e instanceof AnomalyDetectionException) {
-            adExep = (AnomalyDetectionException) e;
-        } else {
-            adExep = new AnomalyDetectionException(detectorId, e);
-        }
-        state.setException(adExep);
+        // AnomalyDetectionException adExep = null;
```

Comment:
Either remove commented code or add descriptive comment about why it is commented.

Reply:
my bad.  reverted https://github.com/opensearch-project/anomaly-detection/commit/daf104a1eca785bdc76e08de5bbbf5bfcc15d24c

Reply:
About the gradle change: We need to use System.getProperty in the Bwc class to check a flag: if the system property is true, then set DISABLE_BWC to true; otherwise, set DISABLE_BWC to false.  It is doable. But the scope is much larger: we need to make tasks like gradle run to accept the property.  Tasks like grade run are defined in Opensearch core and we inherited the build script.

Also, it is not just gradle.  People need it to build domain and create e2e testing.  We need to add the system property to the start up script of Opensearch and managed service.

These stuff is out of AD and hard to control/maintain.  Any other ways to get this to work?

Reply:
@dblock Please also comment if you have any suggestion.